holidays/countries/ukraine.py

```diff
@@ -50,147 +49,118 @@ def _add_with_observed(
                 obs_date = hol_date + td(
                     days=2 if self._is_saturday(hol_date) else days
                 )
-                self[obs_date] = self.tr("Вихідний за %s") % hol_name
+                self[obs_date] = self.tr("%s (вихідний)") % hol_name
 
         # The current set of holidays came into force in 1991
-        # But most holiday days were implemented in 1918
-        if year <= 1917:
+        if year <= 1990:
+            return None
+
+        # There is no holidays in Ukraine during the period of martial law
+        # https://zakon.rada.gov.ua/laws/show/2136-20#n26
+        # law is in force from March 15, 2022
+        if year >= 2023:
             return None
 
         super()._populate(year)
 
-        # New Year's Day
-        if year <= 1929 or year >= 1948:
-            _add_with_observed(date(year, JAN, 1), self.tr("Новий рік"))
+        # New Year's Day.
+        _add_with_observed(date(year, JAN, 1), self.tr("Новий рік"))
 
-        # Christmas Day (Julian calendar)
-        if year >= 1991:
-            _add_with_observed(
-                date(year, JAN, 7),
-                self.tr("Різдво Христове (за юліанським календарем)"),
-            )
+        _add_with_observed(
+            date(year, JAN, 7),
+            # Christmas (Julian calendar).
+            self.tr("Різдво Христове (за юліанським календарем)"),
+        )
 
-        # Women's Day
-        if year >= 1966:
-            _add_with_observed(
-                date(year, MAR, 8), self.tr("Міжнародний жіночий день")
-            )
+        _add_with_observed(
+            date(year, MAR, 8),
+            # International Women's Day.
+            self.tr("Міжнародний жіночий день"),
+        )
+
+        if year >= 2022:
```

Comment:
It seems fair. I'm not sure if we want to loop over and remove MAR+ 2022 holidays instead.

Anyway the comment ref to https://zakon.rada.gov.ua/laws/show/2136-20#n26 would be great.